qemu/tests/virtio_fs_multi_users_access.py

```diff
@@ -0,0 +1,212 @@
+        """
+        error_context.context("Get the ip config from guest.", test.log.info)
+        ipconfig_output = session.cmd_output("ipconfig /all")
+        match = "Connection-specific DNS Suffix  . : %s" % dns
+        if match not in ipconfig_output:
+            test.error("There is NOT related domain adapter found!")
+        eth_name = ipconfig_output.split(match)[0] \
+            .split("Ethernet adapter ")[-1] \
+            .split(":")[0]
+        return eth_name
+
+    def switch_dynamic_ip_to_static_ip(session, eth_name):
+        """
+        Assign a static IP address for the adaptor that
+        can connect to the AD server.
+
+        :param session: the session of guest
+        """
+        error_context.context("Get the config of %s." % eth_name,
+                              test.log.info)
+        net = session.cmd_output("netsh interface ip show config name=\"%s\"" %
+                                 eth_name)
+
+        ipaddr, gateway, dns_server, subnet_mask = "", "", "", ""
+        for line in net.splitlines():
+            if "IP Address: " in line:
+                ipaddr = line.split(":")[-1].lstrip()
+            elif "Default Gateway:" in line:
+                gateway = line.split(":")[-1].lstrip()
+            elif "Subnet Prefix:" in line:
+                subnet_mask = line.split("mask")[-1].lstrip()[:-1]
+        error_context.context("The config will be set to ipaddress:%s, "
+                              "gateway:%s, subnet mask:%s." % (
+                                  ipaddr, gateway, subnet_mask),
+                              test.log.info)
+
+        ip_cmd = "netsh interface ip set address name=\"%s\" source=static " \
+                 "addr=%s mask=%s gateway=%s" % (eth_name, ipaddr,
+                                                 subnet_mask, gateway)
+        session.cmd(ip_cmd)
+
+        dns_cmd = "netsh interface ip set dnsservers \"%s\" static " \
+                  "192.168.0.1 primary" % eth_name
+        session.cmd(dns_cmd)
+
+    def switch_ip_to_dynamic(session, eth_name):
+        if eth_name:
+            restore_ip_cmd = "netsh interface ip set address name=\"%s\" " \
+                             "source=dhcp" % eth_name
+            session.cmd(restore_ip_cmd)
+            restore_dns_cmd = "netsh interface ip set dnsservers name=\"%s\" " \
+                              "source=dhcp" % eth_name
+            session.cmd(restore_dns_cmd)
+
+    add_user_cmd = params.get("add_user_cmd")
+    del_user_cmd = params.get("del_user_cmd")
+    driver_name = params.get("driver_name")
+    fs_target = params.get("fs_target")
+    fs_dest = params.get("fs_dest")
+    os_type = params.get("os_type")
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+
+    username = params.objects("new_user")
+    pwd = params.objects("new_pwd")
+    dns = params.get("dns")
+    eth_name = ""
+
+    try:
+        if not dns:
```

Comment:
I think 'dns' is not a good expression here, how about using ad_user and windows.
if not ad_user and os_type = 'windows'